cmake/StlabUtil.cmake

```diff
@@ -194,7 +196,7 @@ function( stlab_generate_config_file )
     set( STLAB_MAIN_EXECUTOR_LIBDISPATCH TRUE )
   elseif (STLAB_MAIN_EXECUTOR STREQUAL "emscripten")
     set( STLAB_MAIN_EXECUTOR_EMSCRIPTEN TRUE )
-  elseif (STLAB_MAIN_EXECUTOR STREQUAL "qt")
+  elseif (STLAB_MAIN_EXECUTOR MATCHES "^qt[56]$")
     set( STLAB_MAIN_EXECUTOR_QT TRUE )
```

Comment:
I think it would be better to create `STLAB_MAIN_EXECUTOR_QT5` and `STLAB_MAIN_EXECUTOR_QT6` here. The fact that the code for these happens to be the same could be reflected in a `#if STLAB_MAIN_EXECUTOR(QT5) || STLAB_MAIN_EXECUTOR(QT6)` directive in `main_executor.hpp`.

It might be additionally good to have a check in that file that verifying that the major version of Qt that the header provides is consistent with what config.hpp expects. This would ensure we get compile time errors if main_executor.hpp is processed with the include path to Qt6 when Qt5 is what is supposed to be there.